plugin/hooks.go

```diff
@@ -270,4 +271,9 @@ type Hooks interface {
 	//
 	// Minimum server version: 7.0
 	OnCloudLimitsUpdated(limits *model.ProductLimits)
+
+	// UserHasBeenDeactivated is invoked when a user is made inactive.
+	//
+	// Minimum server version:
+	UserHasBeenDeactivated(userID string)
```

Comment:
@ashishbhate The context passed into hooks is the `plugin.Context` struct. In this case, I don't think the `plugin.Context` struct would be all that useful, but we can pass it for consistency and future-proofing. It also seems that many of the methods prefer `*model.User` rather than accepting a user ID

Reply:
@mickmister I'm 2/5 that we should add a `plugin.Context` and use `*model.User`. I'll defer to you though.

Reply:
@ashishbhate @mickmister I added these changes